### ISSUE TITLE
WebSocketApp Blocked on socket.py

### ISSUE THREAD

**Issue by stheakanath** [websocket-client/websocket-client#297]
I created a WebSocketApp instance as shown below:
`
self.ws = websocket.WebSocketApp(self.wssLink, on_open = self.on_open,
							  on_message = self.on_message,
							  on_error = self.on_error,
							  on_close = self.on_close, header ={"user-agent": userAgent})
self.ws.run_forever(origin='https://play.spotify.com')	
`

but after receiving the first message through self.on_message, the websocket app is hung indefinitely. I ran a stacktrace on the system and it gets hung by this line:

`
decoder.py(379):         try:
decoder.py(380):             obj, end = self.scan_once(s, idx)
decoder.py(383):         return obj, end
decoder.py(365):         end = _w(s, end).end()
decoder.py(366):         if end != len(s):
decoder.py(368):         return obj
SXSpotifyWSS.py(77): 		print("Received a message: {}".format(msg))
Received a message: {u'id': 0, u'result': u'ok'}
SXSpotifyWSS.py(78): 		if "id" in msg:
SXSpotifyWSS.py(79): 			print ""

SXSpotifyWSS.py(80): 		if "message" in msg:
_app.py(226):                 if ping_timeout and self.last_ping_tm \
_app.py(198):             while self.sock.connected:
_app.py(199):                 r, w, e = select.select(
_app.py(200):                     (self.sock.sock, ), (), (), ping_timeout)
 --- modulename: socket, funcname: meth
socket.py(228):     return getattr(self._sock,name)(*args)
`

After waiting 1-2 minutes, more messages come in, but the socket closes since there was a blocker.

**Comment by stheakanath** [comment]
Just did a packet sniff and the packets come in right after receiving the message. This is an issue with socket.py where it's getting blocked and websocket-python cannot parse the packet.

**Comment by willmcgugan** [comment]
This is almost certainly an isssue I reported... https://github.com/websocket-client/websocket-client/issues/296

This fork has a fix with worked for us https://github.com/wildfoundry/websocket-client

**Comment by stheakanath** [comment]
Thanks!

**Comment by willmcgugan** [comment]
There may still be issues with that fork. I'd avoid using it in production...

**Comment by tomatpasser** [comment]
I am experiencing the exact same problem - did you find a more robust solution or is the fork stable now? Thanks

**Comment by engn33r** [comment]
I will close this stale issue since I haven't seen any recent issues indicating this is still a problem. If @willmcgugan is correct and this issue is a duplicate of #296, it may have quietly been fixed.
